Sources/Private/Experimental/ValueProviderStore.swift

```diff
@@ -83,7 +78,20 @@ final class ValueProviderStore {
 
   // MARK: Private
 
-  private var valueProviders = [AnimationKeypath: AnyValueProvider]()
+  private var valueProviders = [(keypath: AnimationKeypath, valueProvider: AnyValueProvider)]()
+
+  /// Retrieves the most-recently-registered Value Provider that matches the given keypat
+  private func valueProvider(for keypath: AnimationKeypath) -> AnyValueProvider? {
+    // Iterate backwards though the list of value providers,
+    // so the first match we find is the one that was registered most recently
+    for (registeredKeypath, valueProvider) in valueProviders.reversed() {
+      if registeredKeypath.matches(keypath) {
+        return valueProvider
+      }
+    }
+
+    return nil
```

Comment:
nit: if you want to be concise!
```suggestion
    valueProviders.reversed().first(where: { $0.matches($1) }.valueProvider
```